backend/controllers/chatController.js

```diff
@@ -0,0 +1,154 @@
+const mongoose = require("mongoose");
+const path = require('path');
+const { v4: uuidv4 } = require('uuid');
+const { Chat, Message, Poll  } = require("../models/index");
+
+let connections = {};
+
+async function post_message(req, res) {
+    const { new_message, chat_id } = req.params;
+    update_clients_in_chat(new_message, chat_id);
+    res.send("User post message");
+}
+
+function update_clients_in_chat(new_update, chat_id) {
+    let clients = connections[chat_id];
+    clients.forEach(client => client.response.write(`data: ${JSON.stringify(new_update)}\n\n`));
+}
+
+async function subscribe_chat(req, res) {
+    try {
+        const { chat_id } = req.params;
+        const headers = {
+            'Content-Type': 'text/event-stream',
+            'Connection': 'keep-alive',
+            'Cache-Control': 'no-cache'
+        };
+        res.writeHead(200, headers);
+
+        const chat_buffer = await load_chat(chat_id, Date.now());
+        const data = `data: ${JSON.stringify(chat_buffer)}\n\n`;
+        res.write(data);
+
+        // Use a unique id for each client
+        const clientId = uuidv4(); 
+        // Add the client to the connections object
+        if (!connections[chat_id]) {
+            connections[chat_id] = [];
+        }
+        connections[chat_id].push({
+            id: clientId,
+            response: res
+        });
+
+        req.on("close", () => {
+            // Remove the client from the chat_id's connections
+            connections[chat_id] = connections[chat_id].filter(client => client.id !== clientId);
+            res.end(); // close the connection
+        })
+    } catch(err) {
+        throw new Error(`${err.message}`);
+    }
+}
+
+async function load_chat(chat_id, timestamp, buffer_size=10) {
+    try {
+        // validate chat_id (2 stesps)
+        const is_valid_id = mongoose.Types.ObjectId.isValid(chat_id);
```

Comment:
Why do you need to test for an invalid id? The Chat will just say it doesn't exist.

Reply:
Here's my thought: If we put an invalid chat_id to findById, it will throw a `MongoException: Invalid object ID` which we also have to catch, so I just catch it early with a custom Error but i think this should goes into ChatService as well